lib/gooddata/bricks/middleware/dwh_middleware.rb

```diff
@@ -0,0 +1,28 @@
+# encoding: UTF-8
+#
+# Copyright (c) 2010-2015 GoodData Corporation. All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+require_relative 'base_middleware'
+
+require 'gooddata_datawarehouse' if RUBY_PLATFORM == 'java'
```

Comment:
IMHO the program should gracefully fail when you attempt to use `gooddata_datawarehouse` outside jruby. Else the program will continue in a crippled state and you might not notice.